agents-api/agents_api/activities/task_steps/base_evaluate.py

```diff
@@ -38,10 +38,14 @@ def __init__(self, error, expression, values):
 def _recursive_evaluate(expr, evaluator: SimpleEval):
     if isinstance(expr, str):
         try:
+            if isinstance(expr, str) and expr.startswith("$"):
```

Comment:
this should check for `"$ "` (notice space at the end). Because `$1million` could be a valid string and means something else entirely